icu4j/main/currdata/src/main/java/com/ibm/icu/impl/ICUCurrencyDisplayInfoProvider.java

```diff
@@ -325,7 +325,7 @@ Map<String, String> fetchUnitPatterns() {
             Map<String, String> result = unitPatternsCache;
             if (result == null) {
                 result = new HashMap<>();
-                CurrencySink sink = new CurrencySink(!fallback, CurrencySink.EntrypointTable.CURRENCY_UNIT_PATTERNS);
+                CurrencySink sink = new CurrencySink(false, CurrencySink.EntrypointTable.CURRENCY_UNIT_PATTERNS);
```

Comment:
Aha, I guess I did touch this code...

The `fallback` flag is supposed to prevent falling back to the root bundle. As far as I can tell, that is all it is doing here. I don't see how changing this variable would cause it to find `US-Dollar` when it didn't find it before.

Reply:
`de` doesn't define have that resource at all, and that parameter is keeping us from inheriting the default value from `root`.

I think I ended up getting rid of that parameter altogether on the C side because of problems like this, but didn't want to do anything that drastic this late in the cycle.

Reply:
Which resource? `US-Dollar` definitely doesn't look like something that would be in root. Is it an alias or something not being found?

Reply:
`CurrencyUnitPatterns`.